### ISSUE TITLE
Curly quote character is converted wrongly

### ISSUE THREAD

**Issue by yunhojeon** [jgm/pandoc#7879]
**Explain the problem.**

In Korean, letters can follow immediately after quoted text, like in following markdown sample (test.md):

```
마크다운을 워드로 바꾸려면 'pandoc'을 이용하면 된다.
```

Because smart quote algorithm does not work in this case, pandoc converts both single quote characters into closing curly ones.

```
pandoc -t plain test.md
마크다운을 워드로 바꾸려면 ’pandoc’을 이용하면 된다.
```

As a workaround, I have been using literal left and right (curly) single quote characters (test-curly.md):
```
마크다운을 워드로 바꾸려면 ‘pandoc’을 이용하면 된다.
```

It worked up to ver 2.13 by preserving the quote characters.

```
./pandoc-2.13 -t plain test-curly.md
마크다운을 워드로 바꾸려면 ‘pandoc’을 이용하면 된다.
```

but since ver 2.14 to the latest version, left quote character is converted into right quote character.
```
./pandoc-2.14 -t plain test-curly.md
마크다운을 워드로 바꾸려면 ’pandoc’을 이용하면 된다.
```

I think pandoc should not try to _smarten_ curly quote characters even when _smart_ is enabled.
**Pandoc version?**
ver 2.13, 2.14, 2.17, on Windows and MacOS

**Comment by jgm** [comment]
I think you're mistaken: the left-quote character remains a left-quote character...

**Comment by yunhojeon** [comment]
Are you sure? Left (opening) and right (closing) quote characters look similar on github web page, but if you copy the text into a text editor with some different font, you can see that the first quote character is converted.

**Comment by jgm** [comment]
Yes, I'm pretty sure.  Passing your output above to hexdump:
```
% hexdump
’pandoc’
0000000 e2 80 99 70 61 6e 64 6f 63 e2 80 99 0a         
000000d
```
e2 80 is the UTF-8 encoding of the quotation mark; you can see there are two of them.

**Comment by yunhojeon** [comment]
Maybe I was not clear enough. My markdown source file contains ‘pandoc’ as:

e2 80 98 70 61 6e 64 6f 63 e2 80 99

['e2 80 98' is left single quote in UTF-8](https://www.compart.com/en/unicode/U+2018). 'e2 80 99' is right quote. 

The result from pandoc >= 2.14 is ’pandoc’:

e2 80 99 70 61 6e 64 6f 63 e2 80 99

Both left and right quote chars in .md file is converted to e2 80 99 by pandoc. I think the first one 'e2 80 98' in the .md file should remain unchanged in the output.

**Comment by jgm** [comment]
OK, I understand now. This illustrates the issue without the need for sharp eyes:
```
% pandoc -f markdown-smart --ascii
‘pandoc’을
<p>&#x2018;pandoc&#x2019;&#xC744;</p>
% pandoc -f markdown+smart --ascii
‘pandoc’을
<p>&#x2019;pandoc&#x2019;&#xC744;</p>
```

One easy workaround for issues like this is simply to backslash-escape the open quote.
```
\‘pandoc’을
```

The question is whether we should change smart quote detection so that it is only triggered by straight quotes.  There is something to be said for that.  I think the reason we didn't do that originally is that some formats really wanted quoted sections to be parsed; e.g. docbook uses a `<quote>` element, and this would be lost when you use curly quotes.  Still, a lot of the reasons for doing this dated from days before widespread unicode support.  I think that at this point we could, and probably should, make this change.

**Comment by jgm** [comment]
It's hard to anticipate all the unexpected consequences such a change might have.
Among them:  it may affect punctuation moving in citeproc bibliographies.  At least that's something that would have to be checked.

**Comment by yunhojeon** [comment]
Thanks for the workaround (backslash-escaping) suggestion. I can live with this solution.

Would it be too difficult and/or potentially incompatible with existing use cases if the smart quoting algorithm were modified to work for Korean sentences, by allowing Korean character immediately after closing straight quote?

**Comment by jgm** [comment]
> by allowing Korean character immediately after closing straight quote?

Maybe, I'd have to think about it more.  I assume the same is true for Chinese and Japanese?
Are single quote characters in these languages ever used for anything else besides quotes?  The  reason for the heuristics is that in English, `'` is also used as an apostrophe (e.g. in possessives and certain abbreviations).

**Comment by yunhojeon** [comment]
Probably true for Japanese. Don't know about Chinese. 

Another use case of quote characters in Korean (like in English): we use it for abbreviation. For example, the year 2022 is often written as '22.

**Comment by calliecameron** [comment]
The backslash escape disappears when writing markdown output. E.g. when running pandoc from markdown to markdown multiple times to preprocess a file before finally converting it to the output format, the backslash is present in the original input, but not in the first-stage output - and then when using that output as the input to the second stage, the quote gets flipped, because no backslash is present.

Is it possible to make pandoc preserve backslash escapes when writing markdown?

**Comment by jgm** [comment]
> Is it possible to make pandoc preserve backslash escapes when writing markdown?

No, because the AST doesn't keep track of "concrete syntax" details like whether a character is escaped. The best we could do would be to modify the writer to add backslashes before every quote character, which wouldn't be desirable, I think.
